FIP-0086: Using max ticket (#987)
filecoin-project/FIPs: FIPS/fip-0086.md

Two (or more) messages $m1$ and $m2$ are called _equivocating messages_ if $m1.Sender=m2.Sender \land m1.Instance=m2.Instance \land m1.Value ≠ m2.Value \land m1.MsgType=m2.MsgType \land \texttt(if applicable)\ m1.Round=m2.Round$. We call $m1.Sender$ an _equivocating sender_.

> A set of messages $M$ that does not contain equivocating messages is called _clean_. Participants discard all equivocating messages when forming clean sets. In fact, in the proposed version of GossiPBFT in this document, only one of the equivocating messages needs to be stored (thanks to evidences explicitly validating a message).

Reviewer: ```suggestion
A set of messages $M$ that does not contain equivocating messages is called _clean_. Participants discard all equivocating messages when forming clean sets, so that only one of the equivocating messages needs to be stored (thanks to the explicit evidence).
```

Given this is a spec, I don't think we should refer to "the version proposed in this document" or things like that.

Implementations may optimise this algorithm to treat the reception of an aggregated signature over some (MsgType, Instance, Round, Value) as evidence of a message as if it had received the same tuple of values directly from each participant in the aggregate. This may be used to effectively skip a partially-complete step. In the particular case of a DECIDE message, which carries evidence of a strong quorum of COMMITs for the same round and value, a participant immediately sends its own DECIDE for the same value (copying the evidence) and exits the loop at line 7.

Also, concurrently, we expect that the participant feeds to GossiPBFT chains that are incentive-compatible with EC. To this end, we restrict the set C of candidate values that the participant contributes to deciding to only values that either (i) pass the QUALITY step or (ii) may have been decided by other participants (hence the functin `mightHaveBeenDecided`).

#### Valid messages and evidence

> The $\texttt{Valid}()$ predicate (referred to in lines 10, 22, 34, and 42, and 56) is defined below.

Reviewer 2: ```suggestion
The $\texttt{Valid}()$ predicate (referred to in lines 10, 22, 34, and 42, 56 and 58) is defined below.
```

Author: Thanks. Fixed.
